opal/corelib/array.rb

```diff
@@ -2300,7 +2300,7 @@ def zip(*others, &block)
         }
         others[j] = #{(
           Opal.coerce_to?(`o`, Array, :to_ary) ||
-          Opal.coerce_to!(`o`, Enumerator, :each)
+          Opal.coerce_to!(`o`, Enumerator, :to_enum, :each)
```

Comment:
I'm not sure why this can't be just `Opal.coerce_to!(`o`, Enumerator, :to_enum)`, can you explain?

Reply:
`ruby/spec` checks the given argument as https://github.com/ruby/spec/blob/e489657352f3b9ec3b64a4d20160bd12f336bd3e/core/enumerable/fixtures/classes.rb#L145-L149 🤔

Reply:
Ok, that's fine, passing an enumerator to `Array#zip` should be pretty rare anyway